pygmt/src/grd2cpt.py

```diff
@@ -0,0 +1,190 @@
+    output : str
+        Optional argument to set the file name with extension .cpt to store
+        the generated CPT file. If not given or False (default), saves the CPT
+        as the session current CPT.
+    reverse : str
+        Set this to True or c [Default] to reverse the sense of color
+        progression in the master CPT. Set this to z to reverse the sign of
+        z-values in the color table. Note that this change of z-direction
+        happens before *truncate* and *series* values are used so the latter
+        must be compatible with the changed *z*-range. See also
+        :gmt-docs:`cookbook/features.html#manipulating-cpts`.
+    overrule_bg :
+        Overrule background, foreground, and NaN colors specified in the master
+        CPT with the values of the parameters :gmt-term:`COLOR_BACKGROUND`,
+        :gmt-term:`COLOR_FOREGROUND`, and :gmt-term:`COLOR_NAN` specified in
+        the :gmt-docs:`gmt.conf <gmt.conf>` file or on the command line. When
+        combined with ``background``, only :gmt-term:`COLOR_NAN` is considered.
+    no_bg : bool
+        Do not write out the background, foreground, and NaN-color fields
+        [Default will write them, i.e. ``no_bg=False``].
+    log : bool
+        For logarithmic interpolation scheme with input given as logarithms.
+        Expects input z-values provided via ``series`` to be log10(*z*),
+        assigns colors, and writes out *z*.
+    continuous : bool
+        Force a continuous CPT when building from a list of colors and a list
+        of z-values [Default is None, i.e. discrete values].
+    categorical : bool
+        Do not interpolate the input color table but pick the output colors
+        starting at the beginning of the color table, until colors for all
+        intervals are assigned. This is particularly useful in combination with
+        a categorical color table, like ``cmap='categorical'``.
+    cyclic : bool
+        Produce a wrapped (cyclic) color table that endlessly repeats its
+        range. Note that ``cyclic=True`` cannot be set together with
+        ``categorical=True``.
+    {V}
+    """
+    kind = data_kind(grid)
+    with Session() as lib:
+        if kind == "file":
+            file_context = dummy_context(grid)
+        elif kind == "grid":
+            file_context = lib.virtualfile_from_grid(grid)
+        else:
+            raise GMTInvalidInput("Unrecognized data type: {}".format(type(grid)))
+
+        with file_context as infile:
+            if "W" in kwargs and "Ww" in kwargs:
+                raise GMTInvalidInput(
+                    "Set only categorical or cyclic to True, not both."
+                )
+
+            if "H" in kwargs.keys():  # if output file is set
+                outfile = kwargs.pop("H")
+                if not outfile or not isinstance(outfile, str):
+                    raise GMTInvalidInput("'output' should be a proper file name.")
```

Comment:
@seisman As I understand it, `grd2cpt` can specify can output file that saves the CPT instead of just having it be the session CPT. I figured this was the same process for `grd2cpt` and `makecpt` and I copied the code for this. Is this not the way it would work?

Reply:
To save the CPT to a file, you need to call `grd2cpt ... -H > out.cpt`. See the lines 157 and 158 in `makecpt.py`.

https://github.com/GenericMappingTools/pygmt/blob/921e7a6d7a036ab6ef7975d7449f9627d4f56c05/pygmt/src/makecpt.py#L148-L158

Reply:
I think I fixed it in 201f0f6.